### ISSUE TITLE
Copy API doesn't copy user meta data. [JIRA: RCS-153]

### ISSUE THREAD

**Issue by ksauzz** [basho/riak_cs#1097]
S3 object can contain user meta data which are stored by 'x-amz-meta-*' header on [PUT Object](http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPUT.html), and 'PUT Object - Copy' API should copy these meta data. Riak CS doesn't do that.

This bug is found by [one of ceph tests](https://github.com/basho/s3-tests/blob/a38c056122ef2e8fbd5bc61b387463a16d228a59/s3tests/functional/test_s3.py#L4420-L4434). After the fix, the test can be enabled.
## reproduction

put a object with 2 meta data.

```
% s3curl.pl --id user --put Makefile -- -x localhost:8080  http://test.s3.amazonaws.com/1 -H "x-amz-meta-param1: value1" -H "x-amz-meta-param2: value2" -i
HTTP/1.1 100 Continue

HTTP/1.1 200 OK
Server: Riak CS
ETag: "487d4776ae93cc20feddd887e1226fc8"
Date: Thu, 19 Mar 2015 07:32:04 GMT
Content-Type: text/plain
Content-Length: 0
```

get a object

```
% s3curl.pl --id user -- -x localhost:8080  http://test.s3.amazonaws.com/1 -X HEAD -i
HTTP/1.1 200 OK
x-amz-meta-param2: value2               <---- meta data
x-amz-meta-param1: value1               <---- meta data
Server: Riak CS
Last-Modified: Thu, 19 Mar 2015 07:32:04 GMT
ETag: "487d4776ae93cc20feddd887e1226fc8"
Date: Thu, 19 Mar 2015 07:32:18 GMT
Content-Type: application/octet-stream
Content-Length: 6239
```

copy a object 

```
% s3curl.pl --id user --copySrc test/1 -- -x localhost:8080  http://test.s3.amazonaws.com/2 -H "Content-Length: 0" -i
HTTP/1.1 200 OK
Server: Riak CS
ETag: "487d4776ae93cc20feddd887e1226fc8"
Date: Thu, 19 Mar 2015 07:33:46 GMT
Content-Type: application/xml
Content-Length: 175

<?xml version="1.0" encoding="UTF-8"?><CopyObjectResult><LastModified>2015-03-19T07:33:46.000Z</LastModified><ETag>"487d4776ae93cc20feddd887e1226fc8"</ETag></CopyObjectResult>
```

get a copied object 

```
% s3curl.pl --id user -- -x localhost:8080  http://test.s3.amazonaws.com/2 -X HEAD -i
HTTP/1.1 200 OK
Server: Riak CS
Last-Modified: Thu, 19 Mar 2015 07:33:46 GMT                    <------ no meta data
ETag: "487d4776ae93cc20feddd887e1226fc8"
Date: Thu, 19 Mar 2015 07:34:21 GMT
Content-Type: application/octet-stream
Content-Length: 6239
```

**Comment by shino** [comment]
Quote from AWS doc [1]

> When copying an object, you can preserve most of the metadata (default) or specify new metadata. 
> However, the ACL is not preserved and is set to private for the user making the request. 

[1] PUT Object - Copy - Amazon Simple Storage Service
  http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectCOPY.html

**Comment by shino** [comment]
Will be addressed by #1212

**Comment by Basho-JIRA** [comment]
For release note: this change is part of RCS-257, please refer its comment. 

**_[posted via JIRA by Shunichi Shinohara]**_
